File: 7.Sliding_Window/Python/22.maximum_sub_array.py
def max_subarray_sum(N, arr):
    max_sum = float('-inf')
    current_sum = 0

    for num in arr:
        current_sum += num
        max_sum = max(max_sum, current_sum)
        if current_sum < 0:
            current_sum = 0

    return max_sum

N = 5
arr = [1, 2, 0, 4, 5]
res = max_subarray_sum(N, arr)
print(res)

File: 7.Sliding_Window/Javascript/22.maximum_sub_array.js
function maxSubarraySum(N, arr) {
    let maxSum = -Infinity;
    let currentSum = 0;

    for (let i = 0; i < N; i++) {
        currentSum += arr[i];
        maxSum = Math.max(maxSum, currentSum);
        if (currentSum < 0) {
            currentSum = 0;
        }
    }

    return maxSum;
}

let N = 5;
let arr = [1, 2, 0, 4, 5];
let res = maxSubarraySum(N, arr);
console.log(res);  // Output: 12
